UnitsNet/QuantityFormatter.cs

```diff
@@ -3,186 +3,222 @@
+                    {
+                        throw new FormatException($"The \"{format}\" format string is invalid because the index is out of range.");
+                    }
 
-                        return abbreviations[precisionSpecifier];
-                    case 'V':
-                    case 'v':
-                        return quantity.Value.ToString(formatProvider);
-                    case 'U':
-                    case 'u':
-                        return quantity.Unit.ToString();
-                    case 'Q':
-                    case 'q':
-                        return quantity.QuantityInfo.Name;
-                    case 'S':
-                    case 's':
-                        return ToStringWithSignificantDigitsAfterRadix(quantity, formatProvider, precisionSpecifier);
-                    default:
-                        throw new FormatException($"The {format} format string is not supported.");
+                    return abbreviations[abbreviationIndex];
                 }
-            }
-            else
-            {
-                // Anything else is a standard numeric format string with default unit abbreviation postfix.
-
-                var abbreviations = UnitAbbreviationsCache.Default.GetUnitAbbreviations(quantity.Unit, formatProvider);
-                return string.Format(formatProvider, $"{{0:{format}}} {{1}}", quantity.Value, abbreviations.First());
+                case 'C' or 'c' when int.TryParse(format.Substring(1), out _):
+                    throw new FormatException($"The \"{format}\" (currency) format is not supported.");
+                case 'P' or 'p' when int.TryParse(format.Substring(1), out _):
+                    throw new FormatException($"The \"{format}\" (percent) format is not supported.");
+#endif
             }
         }
+        
+        // Anything else is a standard numeric format string with default unit abbreviation postfix.
+        return FormatWithValueAndAbbreviation(quantity, format, formatProvider);
+    }
 
-        private static string ToStringWithSignificantDigitsAfterRadix<TUnitType>(IQuantity<TUnitType> quantity, IFormatProvider formatProvider, int number) where TUnitType : Enum
-        {
-            string formatForSignificantDigits = UnitFormatter.GetFormat(quantity.Value, number);
-            object[] formatArgs = UnitFormatter.GetFormatArgs(quantity.Unit, quantity.Value, formatProvider, Enumerable.Empty<object>());
-            return string.Format(formatProvider, formatForSignificantDigits, formatArgs);
-        }
+    private static string FormatWithValueAndAbbreviation<TUnitType>(IQuantity<TUnitType> quantity, string format, IFormatProvider formatProvider)
+        where TUnitType : Enum
+    {
+        var abbreviation = UnitsNetSetup.Default.UnitAbbreviations.GetDefaultAbbreviation(quantity.Unit, formatProvider);
+        return string.Format(formatProvider, $"{{0:{format}}} {{1}}", quantity.Value, abbreviation);
+    }
```

Comment:
Another change from the existing behavior (which I think _is necessary_) is that the when in the case of the "standard numeric format" (which is the default for `Volume.FromAuTablespoon(1).ToString()`) we again rely on the default abbreviation (possibly `string.Empty`) instead of throwing, as would be the case if you attempt this on a previous version (e.g. v5):

`Volume.FromAuTablespoon(1).ToString("g"); // FormatException??`